### PR TITLE
feat: add terminal output with ANSI color support

### PR DESCRIPTION
**Pull request by omercnet**
## Summary

Adds ANSI escape code parsing and rendering for terminal output in tool calls, making command-line output colorful and readable.

## Demo

ANSI Output

## Changes

```
┌────────────────────────────────────────────┐
│  🎨  ANSI COLOR SUPPORT  🎨               │
├────────────────────────────────────────────┤
│                                            │
│  \e[32m✓ passed\e[0m → ✓ passed (green)   │
│  \e[31m✗ failed\e[0m → ✗ failed (red)     │
│  \e[1mbold\e[0m     → bold                 │
│                                            │
└────────────────────────────────────────────┘
```

- **Custom ANSI parser** - Pure TypeScript, no external dependencies
- **Full color support** - Standard (30-37), bright (90-97), backgrounds (40-47, 100-107)
- **Text styles** - Bold, dim, italic, underline
- **VS Code integration** - Uses native terminal color variables for theme consistency
- **XSS safe** - HTML is properly escaped before color wrapping

## Files Changed

| File | Purpose |
|------|---------|
| `src/views/webview/main.ts` | ANSI parser + getToolsHtml updates |
| `media/main.css` | 36 ANSI color classes using VS Code vars |
| `src/test/webview.test.ts` | 22 new tests for ANSI functionality |

## Testing

- 127 tests passing (18 new for `ansiToHtml`, 4 for terminal output in tools)
- Handles edge cases: empty strings, no ANSI codes, mixed content, HTML escaping

Closes vscode-acp--oktjg-mjphd3aumgl

### CODE REVIEW

File: src/views/webview/main.ts [hunk 1]
```diff
@@ -61,6 +61,124 @@ export function escapeHtml(str: string): string {
...
+      } else if (ANSI_BACKGROUND[code]) {
+        currentClasses = currentClasses.filter((c) => !isBackgroundClass(c));
+        currentClasses.push(ANSI_BACKGROUND[code]);
+      }
+    }
+
+    lastIndex = match.index + match[0].length;
+  }
+
+  ANSI_ESCAPE_REGEX.lastIndex = 0;
```
[the remark this answers is not in the document]
Comment (omercnet) [reply src/views/webview/main.ts]: This is the correct pattern for reusing a global regex. The reset is necessary because `exec()` updates `lastIndex` and we need a clean state for the next call. The regex is module-scoped and reused across multiple `ansiToHtml` calls.

File: src/views/webview/main.ts [hunk 2]
```diff
@@ -86,8 +204,16 @@ export function getToolsHtml(tools: Record<string, Tool>): string {
           tool.output.length > 500
             ? tool.output.slice(0, 500) + "..."
             : tool.output;
+        const outputHtml = hasAnsiCodes(truncated)
+          ? ansiToHtml(truncated)
+          : escapeHtml(truncated);
+        const terminalClass = hasAnsiCodes(tool.output) ? " terminal" : "";
```
[the remark this answers is not in the document]
Comment (omercnet) [reply src/views/webview/main.ts]: Fixed in 6ab0800 - now caches the result in `hasAnsi` variable.

File: src/views/webview/main.ts [hunk 3]
```diff
@@ -61,6 +61,124 @@ export function escapeHtml(str: string): string {
...
+    if (match.index > lastIndex) {
+      const textContent = escapeHtml(text.slice(lastIndex, match.index));
+      if (currentClasses.length > 0) {
+        result += `<span class="${currentClasses.join(" ")}">${textContent}</span>`;
+      } else {
+        result += textContent;
+      }
+    }
+
+    const codes = match[1].split(";").map((c) => parseInt(c, 10) || 0);
```
[the remark this answers is not in the document]
Comment (omercnet) [reply src/views/webview/main.ts]: Valid edge case, but trailing semicolons like `\x1b[31;m` are malformed ANSI sequences and extremely rare in practice. The current behavior (treating empty as reset) is reasonable and matches common terminal emulator behavior.

File: src/views/webview/main.ts [hunk 4]
```diff
@@ -61,6 +61,124 @@ export function escapeHtml(str: string): string {
...
+          currentClasses.push(styleClass);
+        }
+      } else if (ANSI_FOREGROUND[code]) {
+        currentClasses = currentClasses.filter((c) => !isForegroundClass(c));
+        currentClasses.push(ANSI_FOREGROUND[code]);
+      } else if (ANSI_BACKGROUND[code]) {
+        currentClasses = currentClasses.filter((c) => !isBackgroundClass(c));
+        currentClasses.push(ANSI_BACKGROUND[code]);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (omercnet) [reply src/views/webview/main.ts]: Good point. These codes are rarely used in practice - most tools use code 0 (full reset) instead. Can add support in a follow-up if needed.

File: src/test/webview.test.ts [hunk 5]
```diff
@@ -774,4 +776,191 @@ suite("Webview", () => {
...
+      assert.strictEqual(hasAnsiCodes(""), false);
+    });
+
+    test("returns true for multiple ANSI codes", () => {
+      assert.strictEqual(
+        hasAnsiCodes("\x1b[1;31;42mbold red on green\x1b[0m"),
+        true
+      );
+    });
+  });
```
[the remark this answers is not in the document]
Comment (omercnet) [reply src/test/webview.test.ts]: The 22 vs 18 discrepancy was a typo - the actual count is correct in the tests. Updated description.

File: src/views/webview/main.ts [hunk 6]
```diff
@@ -61,6 +61,124 @@ export function escapeHtml(str: string): string {
...
+
+function isForegroundClass(cls: string): boolean {
+  return (
+    cls.startsWith("ansi-") &&
+    !cls.startsWith("ansi-bg-") &&
+    !cls.startsWith("ansi-bold") &&
+    !cls.startsWith("ansi-dim") &&
+    !cls.startsWith("ansi-italic") &&
+    !cls.startsWith("ansi-underline")
+  );
```
[the remark this answers is not in the document]
Comment (omercnet) [reply src/views/webview/main.ts]: Using a Set would be cleaner but adds allocation overhead for each call. The current approach is simple and fast. Class names are stable (defined in same file) so fragility is low.
